File: CloudBridge-PM/src/CloudBridge/event/PacketReceiveEvent.php
<?php

namespace CloudBridge\event;

use CloudBridge\network\protocol\packet\InvalidPacket;
use CloudBridge\network\protocol\packet\Packet;
use pocketmine\event\Event;

class PacketReceiveEvent extends Event {

    private Packet $packet;
    private bool $invalid = false;

    public function __construct(Packet $packet) {
        $this->packet = $packet;

        if ($packet instanceof InvalidPacket) $this->invalid = true;
    }

    public function getPacket(): Packet {
        return $this->packet;
    }

    public function isInvalid(): bool {
        return $this->invalid;
    }
}

File: CloudBridge-WD/src/main/java/de/r3pt1s/CloudBridge/event/PacketReceiveEvent.java
// This class was created by r3pt1s
package de.r3pt1s.CloudBridge.event;

import de.r3pt1s.CloudBridge.network.protocol.packet.InvalidPacket;
import de.r3pt1s.CloudBridge.network.protocol.packet.Packet;
import dev.waterdog.waterdogpe.event.Event;

public class PacketReceiveEvent extends Event {

    private Packet packet;
    private boolean invalid = false;

    public PacketReceiveEvent(Packet packet) {
        this.packet = packet;
        if (packet instanceof InvalidPacket) invalid = true;
    }

    public boolean isInvalid() {
        return invalid;
    }

    public Packet getPacket() {
        return packet;
    }
}
